desktop-widgets/divelogexportdialog.cpp

```diff
@@ -275,31 +274,77 @@ void DiveLogExportDialog::export_TeX(const char *filename, const bool selected_o
+		int qty_weight;
+		float total_weight;
 
 		if (need_pagebreak)
 			put_format(&buf, "\\vfill\\eject\n");
 		need_pagebreak = true;
 		put_format(&buf, "\\def\\date{%04u-%02u-%02u}\n",
 		      tm.tm_year, tm.tm_mon+1, tm.tm_mday);
 		put_format(&buf, "\\def\\number{%d}\n", dive->number);
-		put_format(&buf, "\\def\\place{%s}\n", site ? site->name : "");
-		put_format(&buf, "\\def\\spot{}\n");
+		put_format(&buf, "\\def\\sitename{%s}\n", site ? site->name : "");
+		put_format(&buf, "\\def\\gpslat{%f}\n", site ? site->latitude.udeg / 1000000.0 : 0);
+		put_format(&buf, "\\def\\gpslon{%f}\n", site ? site->longitude.udeg / 1000000.0 : 0);
+		put_format(&buf, "\\def\\computer{%s}\n", dive->dc.model);
 		put_format(&buf, "\\def\\country{%s}\n", country.toUtf8().data());
-		put_format(&buf, "\\def\\entrance{}\n");
 		put_format(&buf, "\\def\\time{%u:%02u}\n", FRACTION(dive->duration.seconds, 60));
-		put_format(&buf, "\\def\\depth{%u.%01um}\n", FRACTION(dive->maxdepth.mm / 100, 10));
-		put_format(&buf, "\\def\\gasuse{%u.%01ubar}\n", FRACTION(delta_p.mbar / 100, 10));
+		put_format(&buf, "\\def\\maxtemp{%.2f}\n", (dive->maxtemp.mkelvin) ? (dive->maxtemp.mkelvin / 1000.0) - 273.15 : 0);
+		put_format(&buf, "\\def\\mintemp{%.2f}\n", (dive->mintemp.mkelvin) ? (dive->mintemp.mkelvin / 1000.0) - 273.15 : 0);
+		put_format(&buf, "\\def\\watertemp{%.2f}\n", (dive->watertemp.mkelvin) ? (dive->watertemp.mkelvin / 1000.0) - 273.15 : 0);
+		put_format(&buf, "\\def\\airtemp{%.2f}\n", (dive->airtemp.mkelvin) ? (dive->airtemp.mkelvin / 1000.0) - 273.15 : 0);
+		put_format(&buf, "\\def\\maxdepth{%u.%01u m}\n", FRACTION(dive->maxdepth.mm / 100, 10));
+		put_format(&buf, "\\def\\avedepth{%u.%01u m}\n", FRACTION(dive->meandepth.mm / 100, 10));
+
+		//Code block prints start/end press. for all cylinders used, number of cyl. used, SAC, and total delta_p
+		qty_cyl = 0;
+		for (i = 0; i < MAX_CYLINDERS; i++)
+			if (is_cylinder_used(dive, i)){
+				put_format(&buf, "\\def\\cyl%ddescription{%s}\n", i + 1, dive->cylinder[i].type.description);
+				if (dive->cylinder[i].gasmix.o2.permille > 0){ //This code assumes that all gas must have oxygen, so a zero value indicated that the gas in question is AIR
+					put_format(&buf, "\\def\\cyl%dmixO2{%u\%}\n", i + 1, FRACTION(dive->cylinder[i].gasmix.o2.permille, 10));
+					put_format(&buf, "\\def\\cyl%dmixN2{%u\%}\n", i + 1, FRACTION(1000 - dive->cylinder[i].gasmix.o2.permille - dive->cylinder[i].gasmix.he.permille, 10));
+					put_format(&buf, "\\def\\cyl%dmixHe{%u\%}\n", i + 1, FRACTION(dive->cylinder[i].gasmix.he.permille, 10));
+				} else {
+					put_format(&buf, "\\def\\cyl%dmixO2{21\%}\n", i + 1);
+					put_format(&buf, "\\def\\cyl%dmixN2{79\%}\n", i + 1);
+					put_format(&buf, "\\def\\cyl%dmixHe{0\%}\n", i + 1);
+				}
+				delta_p.mbar += dive->cylinder[i].start.mbar - dive->cylinder[i].end.mbar;
+				put_format(&buf, "\\def\\cyl%dstartpress{%u.%01u bar}\n", i + 1, FRACTION(dive->cylinder[i].start.mbar / 100, 10));
```

Comment:
If undefined, I ultimately want it to print a blank field (which will result in no text being printed at that location in the output file).

I'm not sure I like the idea of printing a table directly into the export file, as the benefit of using discrete fields (as opposed to writing string that defines a table directly) is that we can then define in the template what date should be printed in the final pages.  This is why I had an output counter - it will let the template figure out how big of a table should be constructed.